pkg/kubelet/volume/populator/desired_state_of_world_populator.go

```diff
@@ -120,6 +124,32 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 	for _, volumeToMount := range dswp.desiredStateOfWorld.GetVolumesToMount() {
 		if _, podExists :=
 			dswp.podManager.GetPodByUID(volumeToMount.Pod.UID); !podExists {
+			// Once a pod has been deleted from kubelet pod manager, do not
+			// delete it from volume manager. Instead call out to the kubelet
+			// containerRuntime interface to verify all containers in the pod
+			// have been terminated.
+
+			podStatus, podStatusErr := dswp.kubeContainerRuntime.GetPodStatus(
```

Comment:
I'd prefer getting the podStatus through the cache (`kl.podCache`) to avoid hitting the runtime directly.